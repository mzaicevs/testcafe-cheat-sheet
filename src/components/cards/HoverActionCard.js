import React from "react";
import { actions_hover } from "../../snippets";
import CodeCard from "../../base_components/CodeCard";

const HoverActionCard = () => (
  <CodeCard
    cardTitle={actions_hover.title}
    codeSnippet={{
      code: actions_hover.code,
      language: actions_hover.language,
      options: actions_hover.options,
      example: actions_hover.example
    }}
    description={actions_hover.description}
  />
);

export default HoverActionCard;
