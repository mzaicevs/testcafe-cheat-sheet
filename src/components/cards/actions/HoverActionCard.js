import React from "react";
import { actions_hover } from "../../../snippets";
import CodeCard from "../../../base_components/CodeCard";

const HoverActionCard = () => (
  <CodeCard
    cardTitle={actions_hover.title}
    fullDocLink={actions_hover.fullDocLink}
    codeSnippet={{
      code: actions_hover.code,
      language: actions_hover.language,
      options: actions_hover.options,
      example: actions_hover.example
    }}
    description={actions_hover.description}
  />
);

export default HoverActionCard;
