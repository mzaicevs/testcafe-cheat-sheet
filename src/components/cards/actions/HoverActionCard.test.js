import React from "react";
import { shallow } from "enzyme";
import HoverActionCard from "./HoverActionCard";

jest.mock("react-syntax-highlighter/dist/esm/styles/hljs", () => () => {});

describe("ClickActionCard", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<HoverActionCard />);
    expect(wrapper).toMatchSnapshot();
  });
});
