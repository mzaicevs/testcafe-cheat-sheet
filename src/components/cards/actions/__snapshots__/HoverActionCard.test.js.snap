// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClickActionCard matches snapshot 1`] = `
<CodeCard
  cardTitle="Hover Action"
  codeSnippet={
    Object {
      "code": Array [
        "t.hover( selector [, options] )",
      ],
      "example": "import { Selector } from 'testcafe';

  const comboBox = Selector('.combo-box');
  
  fixture \`My fixture\`
      .page \`http://www.example.com/\`;
  
  test('Select combo box value', async t => {
      await t
          .hover(comboBox)
          .click('#i-prefer-both')
          .expect(comboBox.value).eql('Both');
  });",
      "language": "javascript",
      "options": "{
      modifiers: {
          ctrl: Boolean,
          alt: Boolean,
          shift: Boolean,
          meta: Boolean
      },
  
      offsetX: Number,
      offsetY: Number,
      destinationOffsetX: Number,
      destinationOffsetY: Number,
      speed: Number
  }",
    }
  }
  description="Hovers the mouse pointer over a webpage element."
  fullDocLink="https://devexpress.github.io/testcafe/documentation/test-api/actions/hover.html"
/>
`;
